crates/rust-analyzer/src/lsp_utils.rs

```diff
@@ -192,9 +193,9 @@ pub(crate) fn apply_document_changes(
     let mut line_index = LineIndex {
         // the index will be overwritten in the bottom loop's first iteration
         index: Arc::new(ide::LineIndex::new(&text)),
-        // We don't care about line endings or offset encoding here.
+        // We don't care about line endings here.
```

Comment:
That's from https://github.com/rust-lang/rust-analyzer/commit/c8b9ec8e62d9f560a6557496bc4b579019ccb509. But I don't remember the code well enough to say.

Reply:
Yeah, I took a second look here, and it does look like an outright bug.

Reply:
Oh dear, yeah, this one was busted pretty substantially. If we didn't get reports about that, that means that no-one actually uses utf-8 :cry: 

It only worked with VS Code because Code uses utf16.